### PR TITLE
duration metric for http paths

### PR DESCRIPTION
**Pull request by glaslos**
HTTP path duration metric. Slight refactor of handlers. See #2

### CODE REVIEW

File: proxy/proxy.go [hunk 1]
```diff
@@ -39,21 +39,24 @@ var (
...
 		prometheus.CounterOpts{
 			Name: "pushproxy_proxied_requests_total",
 			Help: "Number of http proxy requests.",
-		},
-		[]string{"code"},
+		}, []string{"code"},
 	)
+	httpPathHistogram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "pushprox_http_path_duration",
```
Comment (brian-brazil) [inline proxy/proxy.go]: seconds

Comment (brian-brazil) [reply proxy/proxy.go]: Also, the path is redundant. No need to mention label names in metric names.

Comment (glaslos) [reply proxy/proxy.go]: Fixed

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**